bundles/org.openhab.binding.http/src/main/java/org/openhab/binding/http/internal/converter/RollershutterItemConverter.java

```diff
@@ -0,0 +1,103 @@
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.openhab.binding.http.internal.config.HttpChannelConfig;
+import org.openhab.binding.http.internal.transform.ValueTransformation;
+
+/**
+ * The {@link RollershutterItemConverter} implements {@link org.eclipse.smarthome.core.library.items.RollershutterItem}
+ * conversions
+ *
+ * @author Jan N. Klug - Initial contribution
+ */
+
+@NonNullByDefault
+public class RollershutterItemConverter extends AbstractTransformingItemConverter {
+    private final HttpChannelConfig channelConfig;
+
+    private State state = UnDefType.UNDEF;
+
+    public RollershutterItemConverter(Consumer<State> updateState, Consumer<Command> postCommand,
+            @Nullable Consumer<String> sendCommand, ValueTransformation stateTransformations,
+            ValueTransformation commandTransformations, HttpChannelConfig channelConfig) {
+        super(updateState, postCommand, sendCommand, stateTransformations, commandTransformations, channelConfig);
+        this.channelConfig = channelConfig;
+    }
+
+    @Override
+    public String toString(Command command) {
+        String string = channelConfig.commandToFixedValue(command);
+        if (string != null) {
+            return string;
+        }
+
+        if (command instanceof PercentType) {
+            if (command == PercentType.HUNDRED && !channelConfig.downValue.isEmpty()) {
+                return channelConfig.downValue;
+            } else if (command == PercentType.ZERO && !channelConfig.upValue.isEmpty()) {
+                return channelConfig.upValue;
+            } else {
+                return ((PercentType) command).toString();
+            }
+        }
+
+        throw new IllegalArgumentException("Command type '" + command.toString() + "' not supported");
+    }
+
+    @Override
+    protected @Nullable Command toCommand(String string) {
+        if (string.equals(channelConfig.upValue)) {
+            return UpDownType.UP;
+        } else if (string.equals(channelConfig.downValue)) {
+            return UpDownType.DOWN;
+        } else if (string.equals(channelConfig.moveValue)) {
+            return StopMoveType.MOVE;
+        } else if (string.equals(channelConfig.stopValue)) {
+            return StopMoveType.STOP;
+        }
+
+        return null;
+    }
+
+    @Override
+    public State toState(String string) {
+        State newState = UnDefType.UNDEF;
+
+        try {
+            BigDecimal value = new BigDecimal(string);
+            if (value.compareTo(PercentType.HUNDRED.toBigDecimal()) > 0) {
+                value = PercentType.HUNDRED.toBigDecimal();
+            }
+            if (value.compareTo(PercentType.ZERO.toBigDecimal()) < 0) {
+                value = PercentType.ZERO.toBigDecimal();
+            }
+            newState = new PercentType(value);
+        } catch (NumberFormatException e) {
+        }
+
+        state = newState;
+        return newState;
```

Comment:
```suggestion
        try {
            BigDecimal value = new BigDecimal(string);
            if (value.compareTo(PercentType.HUNDRED.toBigDecimal()) > 0) {
                return PercentType.HUNDRED;
            }
            if (value.compareTo(PercentType.ZERO.toBigDecimal()) < 0) {
                return PercentType.ZERO;
            }
            return new PercentType(value);
        } catch (NumberFormatException e) {
        }
        
        return UnDefType.UNDEF;
```

Reply:
I don't think that is equivalent, the internal state is not updated.

Reply:
Oof, you are right, I was getting a bit too ahead of myself. Leave as is then.